uwb/example/lib/screens/game_screen.dart

```diff
@@ -0,0 +1,382 @@
+import 'dart:async';
+
+import 'package:flutter/material.dart';
+import 'package:uwb/uwb.dart';
+
+import '../game_widgets/ball.dart';
+import '../game_widgets/brick.dart';
+import '../game_widgets/brick_field.dart';
+import '../game_widgets/paddle.dart';
+import '../game_widgets/score.dart';
+import '../game_widgets/screen_overlay.dart';
+
+/// The [BreakoutGame] class.
+///
+/// Contains the code for the breakout game.
+class BreakoutGame extends StatefulWidget {
+  /// The [BreakoutGame] constructor.
+  const BreakoutGame({Key? key}) : super(key: key);
+
+  @override
+  BreakoutGameState createState() => BreakoutGameState();
+}
+
+/// Contains the direction enums of the ball.
+enum Direction {
+  /// Ball goes up.
+  up,
+
+  /// Ball goes down.
+  down,
+
+  /// Ball goes left.
+  left,
+
+  /// Ball goes right.
+  right
+}
+
+/// The [BreakoutGameState] class.
+class BreakoutGameState extends State<BreakoutGame> {
+  final Uwb _plugin = Uwb();
+
+  /// Boolean that checks if the game has started.
+  bool gameHasStarted = false;
+
+  /// Boolean that checks if the player has lost the game.
+  bool playerIsDead = false;
+
+  /// X position of the ball.
+  double ballX = 0;
+
+  /// Y position of the ball.
+  double ballY = 0;
+
+  /// X position of the paddle.
+  double paddleX = 0;
+
+  /// Y position of the paddle.
+  double paddleY = 0.9;
+
+  /// Width of the paddle.
+  double paddleWidth = 0.4;
+
+  /// Height of the paddle.
+  double paddeHeight = 0.03;
+
+  /// Int that keeps track of the score.
+  int score = 0;
+
+  /// Direction that keeps track of the vertical direction of the ball.
+  Direction ballYDirection = Direction.down;
+
+  /// Direction that keeps track of the horizontal direction of the ball.
+  Direction ballXDirection = Direction.left;
+
+  /// List of bricks on the screen.
+  late List<Brick> brickFieldList;
+
+  /// Globalkey of the ball.
+  final GlobalKey<State<StatefulWidget>> ballKey = GlobalKey();
+
+  /// Color of the bricks in the game.
+  Color brickColor = Colors.blueAccent;
+
+  double? _xPosition;
+
+  @override
+  void initState() {
+    initPlatformState();
+    joinHost();
+    brickFieldList = generateBrickFieldList();
+    super.initState();
+  }
+
+  /// Starts the breakout game.
+  void startGame() {
+    if (gameHasStarted == false) {
+      gameHasStarted = true;
+      Timer.periodic(const Duration(milliseconds: 8), (Timer timer) {
+        if (ballY < -0.4) {
+          checkBrickCollision();
+        }
+        moveBall();
+        checkWallCollision();
+        movePlatform();
+        if (isPLayerDead()) {
+          playerIsDead = true;
+          timer.cancel();
+          _showDialog('G A M E  O V E R', Colors.redAccent);
+        }
+        if (score == 20) {
+          timer.cancel();
+          _showDialog('Y O U  W I N', Colors.green);
+        }
+      });
+    }
```

Comment:
Reduce nesting of `if` statements when possible:

```suggestion
    if (gameHasStarted) {
      return;
    }
    
    gameHasStarted = true;
    Timer.periodic(const Duration(milliseconds: 8), (Timer timer) {
      if (ballY < -0.4) {
        checkBrickCollision();
      }
    
      moveBall();
      checkWallCollision();
      movePlatform();
      if (isPLayerDead()) {
        playerIsDead = true;
        timer.cancel();
        _showDialog('G A M E  O V E R', Colors.redAccent);
      }
      if (score == 20) {
        timer.cancel();
        _showDialog('Y O U  W I N', Colors.green);
      }
    });
  }
```